go-controller/pkg/ovn/egressip.go

```diff
@@ -389,8 +389,8 @@ func (oc *DefaultNetworkController) reconcileEgressIPPod(old, new *v1.Pod) (err
 				return err
 			}
 			if !podSelector.Empty() {
-				newMatches := podSelector.Matches(newPodLabels)
-				oldMatches := podSelector.Matches(oldPodLabels)
+				newMatches := new != nil && podSelector.Matches(newPodLabels)
```

Comment:
tiny bit confusing that we are not using oldPod and newPod which is what we use to obtain oldPodLabels and newPodLabes, but makes sense here that we don't want to load matches like "in" && "notin" and anything really if the pod does not exist in the first place either old or new.